app/src/main/java/com/github/se/icebreakrr/ui/sections/Filter.kt

```diff
@@ -0,0 +1,323 @@
+        ageToText = ""
+        ageFromText = ""
+        return // exit early so that there aren't 3 overlapping toasts as the next 2 would also be
+        // called
+      }
+      ageFromText.isNotEmpty() && ageFrom == null -> {
+        Toast.makeText(context, "Please enter a valid 'From' age (13 or older)", Toast.LENGTH_SHORT)
+            .show()
+        ageFromText = "" // Clear invalid input
+      }
+      ageToText.isNotEmpty() && ageTo == null -> {
+        Toast.makeText(context, "Please enter a valid 'To' age (13 or older)", Toast.LENGTH_SHORT)
+            .show()
+        ageToText = "" // Clear invalid input
+      }
+    }
+  }
+
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            title = {
+              Text("Filter", color = Color.White, modifier = Modifier.testTag("FilterTopBarTitle"))
+            },
+            modifier = Modifier.testTag("FilterTopBar"),
+            navigationIcon = {
+              IconButton(
+                  onClick = { navigationActions.goBack() },
+                  modifier = Modifier.testTag("Back Button")) {
+                    Icon(
+                        Icons.AutoMirrored.Filled.ArrowBack,
+                        contentDescription = "Back",
+                        tint = Color.White)
+                  }
+            },
+            colors = TopAppBarDefaults.topAppBarColors(containerColor = IcebreakrrBlue))
+      }) { innerPadding ->
+        Column(modifier = Modifier.fillMaxSize().padding(innerPadding).padding(16.dp)) {
+          Text(
+              "Gender",
+              fontSize = (screenHeight.value * titleFontSizeFactor).sp,
+              modifier = Modifier.padding(vertical = 8.dp).testTag("GenderTitle"))
+
+          Row(
+              modifier = Modifier.fillMaxWidth(),
+              horizontalArrangement = Arrangement.Center,
+              verticalAlignment = Alignment.CenterVertically,
+          ) {
+            GenderButton(
+                selected = manSelected, onClick = { manSelected = !manSelected }, label = "Men")
+            GenderButton(
+                selected = womanSelected,
+                onClick = { womanSelected = !womanSelected },
+                label = "Women")
+            GenderButton(
+                selected = otherSelected,
+                onClick = { otherSelected = !otherSelected },
+                label = "Other")
+          }
+
+          Text(
+              "Age Range",
+              fontSize = (screenHeight.value * titleFontSizeFactor).sp,
+              modifier = Modifier.padding(vertical = 8.dp).testTag("AgeRangeTitle"))
+          val textFieldHeight = screenHeight * textFieldHeightFactor
+          val textFieldWidth = screenWidth * textFieldWidthFactor
+
+          Row(
+              modifier = Modifier.fillMaxWidth().padding(horizontal = 16.dp),
+              horizontalArrangement = Arrangement.Center) {
+                OutlinedTextField(
```

Comment:
Maybe try to modularize better by extracting this composable. This would make the code more readable and easier to maintain.